packages/core/entity/FileSet/index.ts

```diff
@@ -202,7 +202,12 @@ export default class FileSet {
         Object.entries(filterValuesByAnnotation).forEach(([annotation, filterValues]) => {
             // If a filter value is `null` then we need to modify the way we approach filtering
             // it in SQL
-            if (filterValues.length === 0) {
+            if (!!this.excludeFilters?.some((filter) => filter.name === annotation)) {
+                sqlBuilder.where(`"${annotation}" IS NULL`);
```

Comment:
I think I'm missing some context for this change. If this PR is about adding the include/exclude filters, why does FileSet already have the `excludeFilters` and `includeFilters` variables? And why weren't they used in `toQuerySQLBuilder` before?

Reply:
The main logic for include, exclude, and fuzzy filters was added in #238 and remains the same. However, the emphasis for that ticket was to integrate with FES, which we query via URLs and not through our database service (which is how we query other data sources, e.g., CSVs). 
The `toQuerySQLBuilder` is ONLY used by our database service, not by FES, so to prevent #238 from getting more bloated than it already was, I left it to be a separate ticket.

Reply:
Okay, thanks. So it looks like `toQueryString` is for FES and `toQuerySQLBuilder` is for database services? 

On line 176, `FileSet.toQueryString` delegates to `filter.toQueryString`: why not follow the same pattern for `toQuerySQLBuilder` and leave it up to the filter class to decide the correct SQL? If I am reading correctly, this would remove the need for the `excludeFilters` and `includeFilters` variables in `FileSet`

Reply:
fwiw I'm quite sure I had a justification for not delegating to the class which I no longer remember. I think it had to do with the way we're mapping the filters to a grouped list for sql that made it seem like it would be complicated... I'll try it though and see if it comes back to me

Reply:
Whatever you end up doing, I think my feedback for this PR would be to just use a comment to explain the decision

Reply:
I tried to reach a middle ground on this in [11f2e05](https://github.com/AllenInstitute/biofile-finder/pull/262/commits/11f2e05e91f1da314a17794fdda1db8c8c1b23b8). Not 100% sure I fully addressed your concerns though, so lmk if you have more questions